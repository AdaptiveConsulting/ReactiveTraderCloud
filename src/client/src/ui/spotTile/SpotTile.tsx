import * as React from 'react'
import * as classnames from 'classnames'
import { PriceButton, PriceMovementIndicator, TradeNotification } from './'
import * as moment from 'moment'
import './spotTile.scss'
import NotionalContainer from './notional/NotionalContainer';

const SPOT_DATE_FORMAT = 'DD MMM'

// TODO: Move these to types and actions
function replaceWithAction(a: any, b: any): void {
  return
}

interface Notification {
  error: any
  notificationType: 'Trade' | 'Sell' | string
}

interface CurrentSpotPrice {
  ask: any
  bid: any
  priceMovementType: string
  spread: {
    formattedValue: string,
  }
  valueDate: number
}

interface Direction {
  name: 'Buy' | 'Sell' | string
}

const directionBuy = { name: 'Buy' }
const directionSell = { name: 'Sell' }

export interface SpotTileProps {
  canPopout: boolean
  currencyChartIsOpening: boolean
  currencyPair: any
  currentSpotPrice: CurrentSpotPrice
  executionConnected: boolean
  hasNotification: boolean
  isRunningInOpenFin: boolean
  isTradeExecutionInFlight: boolean
  notification: Notification
  notional: number
  priceStale: boolean
  pricingConnected: boolean
  title: string
  executeTrade: (direction: any) => void
  onComponentMount: (id: string) => void
  onPopoutClick: () => void
  undockTile: () => void
  displayCurrencyChart: () => void
}

export default class SpotTile extends React.Component<SpotTileProps, {}> {
  props: SpotTileProps

  componentDidMount() {
    const currencyPair = this.props.currencyPair.currencyPair.symbol
    this.props.onComponentMount(currencyPair)
  }

  render() {
    const {
      canPopout, currencyChartIsOpening, currentSpotPrice, currencyPair, executionConnected,
      hasNotification, isRunningInOpenFin, isTradeExecutionInFlight, notification, priceStale, pricingConnected, title,
      onPopoutClick, undockTile, displayCurrencyChart
    } = this.props

    const notionalInputClass = classnames('spot-tile__notional', { hide: hasNotification })
    const spotDateClass = classnames('spot-tile__delivery', { hide: hasNotification })
    const generatedNotification = hasNotification ? this.createNotification(notification) : null
    const priceComponents = this.createPriceComponents(title, currentSpotPrice, hasNotification)
    const showChartIQIcon = isRunningInOpenFin

    const chartIQIconClassName = classnames({
      'spot-tile__icon--hidden': !showChartIQIcon,
      'glyphicon glyphicon-refresh spot-tile__icon--rotate': currencyChartIsOpening,
      'spot-tile__icon--chart glyphicon glyphicon-stats': !currencyChartIsOpening
    })

    const formattedDate = currentSpotPrice ?
      moment(currentSpotPrice.valueDate).format(SPOT_DATE_FORMAT)
      : ''
    const className = classnames('spot-tile', {
      'spot-tile--stale': (!pricingConnected || priceStale) &&
      !(hasNotification && notification.notificationType === 'Trade'),
      'spot-tile--readonly': !executionConnected,
      'spot-tile--executing': isTradeExecutionInFlight,
      'spot-tile--error': hasNotification && notification.error
    })

    const newWindowClassName = classnames('popout__controls  glyphicon glyphicon-new-window', {
      'spot-tile__icon--tearoff': !canPopout,
      'spot-tile__icon--hidden': canPopout
    })
    const spotTileContent = (
      <div>
        <span className="spot-tile__execution-label">Executing</span>
        {priceComponents}
        <NotionalContainer
          className={notionalInputClass}
          currencyPair={currencyPair}
        />
        <div className={spotDateClass}>
          <span className="spot-tile__tenor">SP</span>
          <span className="spot-tile__delivery-date">. {formattedDate}</span>
        </div>
      </div>
    )
    return (
      <div className={className}>
        <div className="spot-tile__container">
          <div className="spot-tile__controls">
            <i className={chartIQIconClassName}
               onClick={() => displayCurrencyChart()}/>
            <i className={newWindowClassName}
               onClick={() => onPopoutClick()}/>
            <i className="popout__undock spot-tile__icon--undock glyphicon glyphicon-log-out"
               onClick={() => undockTile()}/>
          </div>
          {!hasNotification ? spotTileContent : generatedNotification}
        </div>
      </div>
    )
  }

  createPriceComponents(title: string, currentSpotPrice: CurrentSpotPrice, hide: boolean) {
    if (currentSpotPrice === null) return null

    const pricingContainerClass = classnames({ hide })

    return (
      <div className={pricingContainerClass}>
        <span className="spot-tile__symbol">{title}</span>
        <PriceButton
          className="spot-tile__price spot-tile__price--bid"
          direction={directionSell}
          onExecute={() => this.props.executeTrade(createTradeRequest(directionSell, this.props.currencyPair.currencyPair.symbol, this.props.currentSpotPrice.bid, this.props.notional, this.props.currencyPair.currencyPair.base, this.props))}
          rate={currentSpotPrice.bid}/>
        <div className="spot-tile__price-movement">
          <PriceMovementIndicator
            priceMovementType={currentSpotPrice.priceMovementType}
            spread={currentSpotPrice.spread}/>
        </div>
        <PriceButton
          className="spot-tile__price spot-tile__price--ask"
          direction={directionBuy}
          onExecute={() => this.props.executeTrade(createTradeRequest(directionBuy, this.props.currencyPair.currencyPair.symbol, this.props.currentSpotPrice.ask, this.props.notional, this.props.currencyPair.currencyPair.base, this.props))}
          rate={currentSpotPrice.ask}/>
      </div>
    )
  }

  createNotification(notification: any) {
    if (notification.notificationType === 'Trade') {
      return (
        <TradeNotification
          className="spot-tile__trade-summary"
          notification={notification}
          onDismissedClicked={() => replaceWithAction('tradeNotificationDismissed', {})}/>
      )
    } else if (notification.notificationType === 'Text') {
      return (
        <div className="spot-tile__notification-message">{notification.message}</div>
      )
    } else {
      throw new Error(`Unknown notification type ${notification.notificationType}`)
    }
  }
}

const createTradeRequest = (direction: Direction, currencyPair: string, spotRate: any, notional: number, currencyBase: string, props: any) => {
  return {
    CurrencyPair: currencyPair,
    SpotRate: spotRate.rawRate,
    Direction: direction.name,
    Notional: notional,
    DealtCurrency: currencyBase
  }
}
