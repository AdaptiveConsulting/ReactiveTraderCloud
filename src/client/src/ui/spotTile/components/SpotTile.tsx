import numeral from 'numeral'
import React, { PureComponent } from 'react'
import { Flex } from 'rt-components'
import { CurrencyPair, Direction } from 'rt-types'
import { styled } from 'rt-util'
import { spotDateFormatter } from '../model/dateUtils'
import { SpotTileData } from '../model/spotTileData'
import NotionalInput from './notional'
import PriceControls from './PriceControls'
import { DeliveryDate, TileBaseStyle, TileSymbol } from './Styled'

const SpotTileStyle = styled(TileBaseStyle)`
  background-color: ${({ theme: { background } }) => background.backgroundSecondary};
`

const SpotTileWrapper = styled('div')`
  position: relative;
  min-height: 150px;
  height: 100%;
`

export interface Props {
  currencyPair: CurrencyPair
  spotTileData: SpotTileData
  executeTrade: (direction: Direction, notional: number) => void
}

interface State {
  notional: string
}

export default class SpotTile extends PureComponent<Props, State> {
  state = {
    notional: '1000000'
  }

  updateNotional = (notional: string) => this.setState({ notional })

  executeTrade = (direction: Direction) => this.props.executeTrade(direction, numeral(this.state.notional).value())

  render() {
    const { currencyPair, spotTileData, children } = this.props
    const { notional } = this.state

    const priceData = spotTileData && spotTileData.price
    const spotDate = priceData && spotDateFormatter(priceData.valueDate, false).toUpperCase()

    return (
      <SpotTileWrapper className="spot-tile-container">
        <SpotTileStyle className="spot-tile">
          <Flex direction="column" justifyContent="space-between" height="100%">
            <Flex alignItems="center" justifyContent="space-between">
              <TileSymbol>{`${currencyPair.base}/${currencyPair.terms}`}</TileSymbol>
              <DeliveryDate className="delivery-date">{spotDate && `SPT (${spotDate})`} </DeliveryDate>
            </Flex>
            <PriceControls executeTrade={this.executeTrade} priceData={priceData} currencyPair={currencyPair} />
            <NotionalInput
              notional={notional}
              currencyPairSymbol={currencyPair.base}
              updateNotional={this.updateNotional}
            />
          </Flex>
        </SpotTileStyle>
        {children}
      </SpotTileWrapper>
    )
  }
}
