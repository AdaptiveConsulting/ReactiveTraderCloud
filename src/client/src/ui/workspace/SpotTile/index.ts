export { default } from './SpotTile'
export { default as PriceButton } from './PriceButton/'
export { default as PriceMovementIndicator } from './PriceMovementIndicator';
// export { default as NotionalInput } from './notionalInput.jsx';
// export { default as TradeNotification } from './tradeNotification.jsx';
