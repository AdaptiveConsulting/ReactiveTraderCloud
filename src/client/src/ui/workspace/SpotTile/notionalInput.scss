@import "variables.scss";

.notional__currency-pair {
  float: left;
  padding-right: 10px;
  padding-bottom: 1px;
}

.notional__size-input {
  float: left;
  padding: 0 0 0 3px;
  background: transparent;
  border: none;
  border-bottom: 1px solid transparent;
  color: inherit;
  font-size: inherit;
  font-family: inherit;
  width: 100px;
  &:hover,
  &:focus {
    border-bottom: 1px solid $primary-color;
  }
  &:focus {
    outline: none;
    background-color: #1d2027;
    color: white;
  }
  &::selection {
    background-color: #00A8CC;
  }
}
