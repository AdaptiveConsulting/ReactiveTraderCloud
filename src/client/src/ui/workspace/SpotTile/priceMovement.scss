@import "variables.scss";

.price-movement {
  position: relative;
  height: 50px;
}

.price-movement__value,
.price-movement__icon--up,
.price-movement__icon--down {
  position: absolute;
  left: 0;
  right: 0;
  margin: auto;
}

.price-movement__value {
  top: 14px;
  color: $spot-tile-primary-color;
  font-size: 14px;
}

.price-movement__icon--inactive {
  display: inherit;
  color: rgba(255, 255, 255, .1) !important;
}

.price-movement__icon--up {
  color: #81d31c;
  font-size: 12px;
  top: 7px;
}

.price-movement__icon--down {
  color: #d90a0a;
  font-size: 12px;
  bottom: 7px;
}

