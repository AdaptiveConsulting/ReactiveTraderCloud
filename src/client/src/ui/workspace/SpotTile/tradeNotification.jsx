import React from 'react';
import { TradeExecutionNotification } from '../model';
import { TradeStatus } from '../../../services/model';
import classnames from 'classnames';
import './tradeNotification.scss';

export default class TradeNotification extends React.Component {

  static propTypes = {
    className: React.PropTypes.string,
    tradeExecutionNotification: React.PropTypes.object.isRequired,
    onDismissedClicked: React.PropTypes.func
  };

  render(){
    let { tradeExecutionNotification, onDismissedClicked, className } = this.props;

    if(tradeExecutionNotification.hasError) {
      let classNames = classnames(
        'trade-notification',
        'trade-notification--error',
        className
      );
      return (<div className={classNames}>
        {tradeExecutionNotification.error}. The execution status is unknown. Please contact your sales rep.
        <a href='#' className='trade-notification__button--dismiss' onClick={onDismissedClicked}>Done</a>
      </div>);
    } else {
      let classNames = classnames(
        'trade-notification',
        className,
        {
          'trade-notification--rejected': tradeExecutionNotification.status === TradeStatus.Rejected
        }
      );
      return (
        <div className={classNames}>
          <span className='trade-notification__trade-status'>{tradeExecutionNotification.status.name}</span>
          <ul className='trade-notification__summary-items'>
            <li className='trade-notification__summary-item trade-notification__summary-item--direction'>{tradeExecutionNotification.direction} </li>
            <li className='trade-notification__summary-item trade-notification__summary-item--notional'>{tradeExecutionNotification.dealtCurrency} {tradeExecutionNotification.notional}</li>
            <li className='trade-notification__summary-item trade-notification__summary-item--currency'><span className='trade-notification__label--versus'>vs </span>{tradeExecutionNotification.termsCurrency}</li>
          </ul>
          <div className='trade-notification__details-items-container'>
            <ul className='trade-notification__details-items trade-notification__details-items--rate'>
              <li className='trade-notification__details-item trade-notification__details-item--label'>Rate</li>
              <li className='trade-notification__details-item trade-notification__details-item--value'>{tradeExecutionNotification.spotRate}</li>
            </ul>
            <ul className='trade-notification__details-items trade-notification__details-items--date'>
              <li className='trade-notification__details-item trade-notification__details-item--label'>Date</li>
              <li className='trade-notification__details-item trade-notification__details-item--value'>{tradeExecutionNotification.formattedValueDate}</li>
            </ul>
            <ul className='trade-notification__details-items trade-notification__details-items--trade-id'>
              <li className='trade-notification__details-item trade-notification__details-item--label'>Trade ID </li>
              <li className='trade-notification__details-item trade-notification__details-item--value'>{tradeExecutionNotification.tradeId}</li>
            </ul>
          </div>
          <a href='#' className='trade-notification__button--dismiss' onClick={onDismissedClicked}><i className='trade-notification__button--dismiss-icon fa fa-share' ></i></a>
        </div>
      );
    }
  }
}
