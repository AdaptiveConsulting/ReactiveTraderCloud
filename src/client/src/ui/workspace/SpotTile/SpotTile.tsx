import * as React from 'react'
import * as classnames from 'classnames'
import { PriceMovementIndicator, PriceButton, NotionalInput, TradeNotification } from './'
import * as moment from 'moment'
import './spotTile.scss'

const SPOT_DATE_FORMAT = 'DD MMM'

// TODO: Move these to types and actions
function replaceWithAction(a: any, b: any): void {
  return
}
type NotificationType = 'Trade' | 'Sell'
interface Notification {
  error: any
  // tslint:disable-next-line:trailing-comma
  notificationType: NotificationType
}
interface CurrentSpotPrice {
  ask: any
  bid: any
  priceMovementType: string
  spread: {
    // tslint:disable-next-line:trailing-comma
    formattedValue: string
  }
  valueDate: number
}
interface Direction {
  name: 'Buy' | 'Sell' | string
}
const directionBuy = { name: 'Buy' }
const directionSell = { name: 'Buy' }

export interface SpotTileProps {
  canPopout: boolean
  currencyChartIsOpening: boolean
  currencyPair: {
    symbol: string
    // tslint:disable-next-line:trailing-comma
    base: number
  }
  currentSpotPrice: CurrentSpotPrice
  executionConnected: boolean
  hasNotification: boolean
  isRunningInOpenFin: boolean
  isTradeExecutionInFlight: boolean
  maxNotional: number
  notification: Notification
  notional: number
  priceStale: boolean
  pricingConnected: boolean
  title: string
}

export default class SpotTile extends React.Component<SpotTileProps, {}> {
  props: SpotTileProps

  render() {
    const { canPopout, currencyChartIsOpening, currentSpotPrice, currencyPair, executionConnected, 
      hasNotification, isRunningInOpenFin, isTradeExecutionInFlight, maxNotional, notification, 
      notional, priceStale, pricingConnected, title } = this.props

    const notionalInputClass = classnames('spot-tile__notional', { hide: hasNotification })
    const spotDateClass = classnames('spot-tile__delivery', { hide: hasNotification })
    const generatedNotification = hasNotification ? this.createNotification(notification) : null
    const priceComponents = this.createPriceComponents(title, currentSpotPrice, !hasNotification)
    const showChartIQIcon = isRunningInOpenFin

    const chartIQIconClassName = classnames({
      'spot-tile__icon--hidden': !showChartIQIcon,
      'glyphicon glyphicon-refresh spot-tile__icon--rotate': currencyChartIsOpening,
      'spot-tile__icon--chart glyphicon glyphicon-stats': !currencyChartIsOpening,
    })

    const formattedDate = currentSpotPrice ?
      moment(currentSpotPrice.valueDate).format(SPOT_DATE_FORMAT)
      : ''
    const className = classnames('spot-tile', {
      'spot-tile--stale': (!pricingConnected || priceStale) &&
        !(hasNotification && notification.notificationType === 'Trade'),
      'spot-tile--readonly': !executionConnected,
      'spot-tile--executing': isTradeExecutionInFlight,
      'spot-tile--error': hasNotification && notification.error,
    })

    const newWindowClassName = classnames('popout__controls  glyphicon glyphicon-new-window', {
      'spot-tile__icon--tearoff': !canPopout,
      'spot-tile__icon--hidden': canPopout,
    })
    const spotTileContent = (
      <div>
        <span className="spot-tile__execution-label">Executing</span>
        {priceComponents}
        <NotionalInput
          className={notionalInputClass}
          notional={notional}
          onChange={notional => replaceWithAction('notionalChanged', { notional })}
          maxValue={maxNotional}
          currencyPair={currencyPair}/>
        <div className={spotDateClass}>
          <span className="spot-tile__tenor">SP</span>
          <span className="spot-tile__delivery-date">. {formattedDate}</span>
        </div>
      </div>
    )
    return (
      <div className={className}>
        <div className="spot-tile__container">
          <div className="spot-tile__controls">
            <i className={chartIQIconClassName}
               onClick={() => this.displayCurrencyChart()}/>
            <i className={newWindowClassName}
               onClick={() => replaceWithAction('popOutTile', {})}/>
            <i className="popout__undock spot-tile__icon--undock glyphicon glyphicon-log-out"
               onClick={() => replaceWithAction('undockTile', {})}/>
          </div>
          {!hasNotification ? spotTileContent : generatedNotification}
        </div>
      </div>
    )
  }

  displayCurrencyChart() {
    replaceWithAction('displayCurrencyChart', {})
  }

  createPriceComponents(title: string, currentSpotPrice: CurrentSpotPrice, hide: boolean) {
    if (currentSpotPrice === null) return null

    const pricingContainerClass = classnames({ hide })

    return (
      <div className={pricingContainerClass}>
        <span className="spot-tile__symbol">{title}</span>
        <PriceButton
          className="spot-tile__price spot-tile__price--bid"
          direction={directionSell}
          onExecute={() => this.executeTrade(directionSell)}
          rate={currentSpotPrice.bid}/>
        <div className="spot-tile__price-movement">
          <PriceMovementIndicator
            priceMovementType={currentSpotPrice.priceMovementType}
            spread={currentSpotPrice.spread}/>
        </div>
        <PriceButton
          className="spot-tile__price spot-tile__price--ask"
          direction={directionBuy}
          onExecute={() => this.executeTrade(directionBuy)}
          rate={currentSpotPrice.ask}/>
      </div>
    )
  }

  executeTrade(direction: Direction) {
    if (this.props.executionConnected) {
      replaceWithAction('executeTrade', { direction })
    }
  }

  createNotification(notification: any) {
    if (notification.notificationType === 'Trade') {
      return (
        <TradeNotification
          className="spot-tile__trade-summary"
          tradeExecutionNotification={notification}
          onDismissedClicked={() => replaceWithAction('tradeNotificationDismissed', {})}/>
      )
    } else if (notification.notificationType === 'Text') {
      return (
        <div className="spot-tile__notification-message">{notification.message}</div>
      )
    } else {
      throw new Error(`Unknown notification type ${notification.notificationType}`)
    }
  }
}
