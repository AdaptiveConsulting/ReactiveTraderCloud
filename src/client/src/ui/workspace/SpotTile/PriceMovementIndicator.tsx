import * as React from 'react'
import * as classnames from 'classnames'
import './priceMovement.scss'

const priceMovementTypes = {
  Up: 'Up',
  Down: 'Down',
}

export interface PriceMovementIndicatorProps {
  priceMovementType: string
  spread: {
    formattedValue: string,
  }
}

const PriceMovementIndicator = (props: PriceMovementIndicatorProps) => {

  const { priceMovementType, spread } = props

  const upDirectionClassName = classnames(
    'price-movement__icon--up fa fa-lg',
    {
      'fa-caret-up': priceMovementType === priceMovementTypes.Up,
    },
    {
      'price-movement__icon--inactive': priceMovementType === priceMovementTypes.Down,
    },
  )

  const downDirectionClassName = classnames(
    'price-movement__icon--down fa fa-lg',
    {
      'fa-caret-down': priceMovementType === priceMovementTypes.Down,
    },
    {
      'price-movement__icon--inactive': priceMovementType === priceMovementTypes.Up,
    },
  )

  return (
    <div>
      <div className="price-movement">
        <i className={upDirectionClassName}></i>
        <span className="price-movement__value">{spread.formattedValue}</span>
        <i className={downDirectionClassName}></i>
      </div>
    </div>
  )
}

export default PriceMovementIndicator
