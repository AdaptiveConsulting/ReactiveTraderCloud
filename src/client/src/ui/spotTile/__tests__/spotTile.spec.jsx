/* tslint:disable */

import * as React from 'react'
import SpotTile from '../SpotTile'
import * as renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'

// replace with getSpotTileProps from storybook
const spotTileProps = {
  canPopout: true,
  currencyChartIsOpening: false,
  currencyPair: {
    symbol: 'GBP',
    base: 'GBP',
  },
  currentSpotPrice: {
    ask: getButtonProps('ask', () => undefined).rate,
    bid: getButtonProps('ask', () => undefined).rate,
    priceMovementType: 'Up',
    spread: {
      formattedValue: '-1.23',
    },
    valueDate: 1234436547,
  },
  executionConnected: true,
  hasNotification: false,
  isRunningInOpenFin: false,
  isTradeExecutionInFlight: false,
  maxNotional: 5000000,
  // notification: {
  //   error: null,
  //   notificationType: 'Trade',
  // },
  notional: 500,
  priceStale: false,
  title: 'GBP / USD',
}

function getButtonProps (type: string, action: any) {
  const classNameType = type === 'Sell' ? 'bid' : 'ask'

  return {
    className: `spot-tile__price spot-tile__price--${classNameType}`,
    direction: type,
    rate: {
      pips: 5,
      bigFigure: 100,
      pipFraction: 8,
      rawRate: 123,
    },
    onExecute: action('buy clicked'),
  }
}

test('SpotTile renders correct markup', () => {
  const shallowRenderer = new ShallowRenderer()
  shallowRenderer.render(
    <SpotTile {...spotTileProps} />
  )

  expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
})
