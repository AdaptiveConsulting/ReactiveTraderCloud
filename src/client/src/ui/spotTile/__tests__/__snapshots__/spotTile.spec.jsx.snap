// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotTile renders correct markup 1`] = `
<div
  className="spot-tile"
>
  <div
    className="spot-tile__container"
  >
    <div
      className="spot-tile__controls"
    >
      <i
        className="spot-tile__icon--hidden spot-tile__icon--chart glyphicon glyphicon-stats"
        onClick={[Function]}
      />
      <i
        className="popout__controls  glyphicon glyphicon-new-window spot-tile__icon--hidden"
        onClick={[Function]}
      />
      <i
        className="popout__undock spot-tile__icon--undock glyphicon glyphicon-log-out"
        onClick={[Function]}
      />
    </div>
    <div>
      <span
        className="spot-tile__execution-label"
      >
        Executing
      </span>
      <div
        className=""
      >
        <span
          className="spot-tile__symbol"
        >
          GBP / USD
        </span>
        <PriceButton
          className="spot-tile__price spot-tile__price--bid"
          direction={
            Object {
              "name": "Sell",
            }
          }
          onExecute={[Function]}
          rate={
            Object {
              "bigFigure": 100,
              "pipFraction": 8,
              "pips": 5,
              "rawRate": 123,
            }
          }
        />
        <div
          className="spot-tile__price-movement"
        >
          <PriceMovementIndicator
            priceMovementType="Up"
            spread={
              Object {
                "formattedValue": "-1.23",
              }
            }
          />
        </div>
        <PriceButton
          className="spot-tile__price spot-tile__price--ask"
          direction={
            Object {
              "name": "Buy",
            }
          }
          onExecute={[Function]}
          rate={
            Object {
              "bigFigure": 100,
              "pipFraction": 8,
              "pips": 5,
              "rawRate": 123,
            }
          }
        />
      </div>
      <Connect(NotionalContainer)
        className="spot-tile__notional"
        currencyPair={
          Object {
            "base": "GBP",
            "symbol": "GBP",
          }
        }
      />
      <div
        className="spot-tile__delivery"
      >
        <span
          className="spot-tile__tenor"
        >
          SP
        </span>
        <span
          className="spot-tile__delivery-date"
        >
          . 
          15 Jan
        </span>
      </div>
    </div>
  </div>
</div>
`;
